examples/cube/main.rs

```diff
@@ -18,39 +18,45 @@ extern crate gfx;
 extern crate gfx_window_glfw;
 extern crate glfw;
 
-use cgmath::FixedArray;
-use cgmath::{Matrix, Point3, Vector3};
-use cgmath::{Transform, AffineMatrix3};
-use gfx::attrib::Floater;
-use gfx::traits::{Factory, Stream, ToIndexSlice, ToSlice, FactoryExt};
+use gfx::format::{I8Scaled, U8Norm};
 
 // Declare the vertex format suitable for drawing.
 // Notice the use of FixedPoint.
-gfx_vertex!( Vertex {
-    a_Pos@ pos: [Floater<i8>; 3],
-    a_TexCoord@ tex_coord: [Floater<u8>; 2],
+gfx_vertex_struct!( Vertex {
+    pos: [I8Scaled; 3] = "a_Pos",
+    tex_coord: [I8Scaled; 2] = "a_TexCoord",
 });
 
 impl Vertex {
-    fn new(p: [i8; 3], t: [u8; 2]) -> Vertex {
+    fn new(p: [i8; 3], t: [i8; 2]) -> Vertex {
         Vertex {
-            pos: Floater::cast3(p),
-            tex_coord: Floater::cast2(t),
+            pos: I8Scaled::cast3(p),
+            tex_coord: I8Scaled::cast2(t),
         }
     }
 }
 
-// The shader_param attribute makes sure the following struct can be used to
-// pass parameters to a shader.
-gfx_parameters!( Params {
-    u_Transform@ transform: [[f32; 4]; 4],
-    t_Color@ color: gfx::shade::TextureParam<R>,
+gfx_pipeline_init!( PipeData PipeMeta PipeInit {
```

Comment:
Thanks for trying to tackle this problem! I iterated on the separate module idea earlier:
> I tried a different approach - to create a new internal module with a user-given name and then just use some predefined simple names inside. However, this turns out to be playing very badly with user-given parameters, which the user expects to be visible, while jumping into a sub-module instantly changes the visibility scope...

However, I didn't consider/try doing `use super::*;` at the top of the new module. Perhaps, this would work? As a side effect it may require dependent types (vertex structures) to be declared as public or re-exported as public.